src/components/PostDetails.tsx

```diff
@@ -1,100 +1,75 @@
-import React, { useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as commentsActions } from '../features/commentsSlice';
+
+export const PostDetails = () => {
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const {
+    items: comments,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(commentsActions.setLoaded(false));
+    dispatch(commentsActions.setHasError(false));
     setVisible(false);
 
     commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .getPostComments(selectedPost?.id || 0)
+      .then(commentsFromApi =>
+        dispatch(commentsActions.setComments(commentsFromApi)),
+      ) // save the loaded comments
+      .catch(() => dispatch(commentsActions.setHasError(true))) // show an error when something went wrong
```

Comment:
```suggestion
      .catch(() => dispatch(commentsActions.setHasError(true)))
```
it's clear from your code - no need to add a lot of comments. Comments in JS are necessary to explain some uncommon approach or difficult logic. Because code itself is quite self-explanatory (if it's a good code with meaningful variable names)